src/main/scala/com/sky/kafka/configurator/TopicConfigurationParser.scala

```diff
@@ -4,19 +4,21 @@ import java.io.{Reader => JReader}
 
 import cats.instances.either._
 import cats.instances.list._
+import cats.syntax.either._
 import cats.syntax.traverse._
 import io.circe
 import io.circe.generic.AutoDerivation
 import io.circe.yaml.parser._
 import io.circe.{Decoder, DecodingFailure, Json}
 
 import scala.collection.immutable.ListMap
+import scala.util.Try
 
 object TopicConfigurationParser extends AutoDerivation {
 
   def apply(topicConfigReader: JReader): Either[circe.Error, List[Topic]] =
     for {
-      ymlAsJson <- parse(topicConfigReader)
+      ymlAsJson <- Try(parse(topicConfigReader)).fold(_ => Json.obj().asRight, identity)
```

Comment:
Is it worth logging any files that are empty as a warning? Incase it is unexpected?

Reply:
Maybe. One issue with that might be all the logs for this app are collected from one place at the moment and then only logged when the application completes I think. So adding logs from here would probably involve some restructuring which should maybe be a different PR.